packages/app/src/app/components/StripeMessages/FreeViewOnlyStripe.tsx

```diff
@@ -1,25 +1,39 @@
 import React from 'react';
+import styled from 'styled-components';
 import track from '@codesandbox/common/lib/utils/analytics';
 import { MessageStripe, Text } from '@codesandbox/components';
+import { useActions } from 'app/overmind';
 import { useWorkspaceAuthorization } from 'app/hooks/useWorkspaceAuthorization';
 
+const LinkButton = styled.button`
+  border: none;
+  background: none;
+  padding: 0;
+  color: inherit;
+  font-family: 'Inter', sans-serif;
+  text-decoration: underline;
+
+  &:hover {
+    color: white;
+  }
+`;
+
 export const FreeViewOnlyStripe = () => {
+  const { sandboxPrivacyChanged } = useActions().workspace;
   const { isTeamAdmin } = useWorkspaceAuthorization();
 
+  const changeToPublic = () => {
+    sandboxPrivacyChanged({ privacy: 0, source: 'editor-view-only-stripe' });
```

Comment:
I would rename this to `changeSandboxPrivacy` or even better, since the action is a big if/else statement, we can have a `makeSandboxPublic` and a `makeSandboxPrivate` (which can include the unlisted state)

Reply:
We discussed this on Slack and don't want to blow up the scope for now.